Source/WebKit/UIProcess/Cocoa/GPUProcessProxyCocoa.mm

```diff
@@ -89,6 +95,53 @@
     return SandboxExtension::createHandlesForMachLookup({ "com.apple.gputools.service"_s, }, std::nullopt);
 }
 
+RefPtr<XPCEventHandler> GPUProcessProxy::xpcEventHandler() const
+{
+    return adoptRef(new GPUProcessProxy::XPCEventHandler(*this));
+}
+
+bool GPUProcessProxy::XPCEventHandler::handleXPCEvent(xpc_object_t event) const
+{
+    if (!m_gpuProcess)
+        return false;
+
+    if (!event || xpc_get_type(event) == XPC_TYPE_ERROR)
+        return false;
+
+    auto messageName = String::fromUTF8(xpc_dictionary_get_string(event, XPCEndpoint::xpcMessageNameKey));
+    if (messageName.isEmpty())
+        return false;
+
+#if USE(EXTENSIONKIT)
+    if (messageName == String::fromUTF8(layerHostingContextChangedMessage)) {
+        auto mediaPlayerIdentifier = xpc_dictionary_get_uint64(event, mediaplayerIdentifierKey);
+        auto webProcessIdentifier = xpc_dictionary_get_uint64(event, webProcessIdentifierKey);
+        RefPtr<WebProcessProxy> webProcess = WebProcessProxy::processForIdentifier(WebCore::ProcessIdentifier(webProcessIdentifier));
+        if (webProcess) {
```

Comment:
Nit: You could scope `webProcess` to the `if () { ...` statement.

Reply:
Done!